### PR TITLE
I18nRubyGem: new parser

### LINKED ISSUE #3523: YAML: extract keys (with scope) for I18n Ruby Gem
The name of the parser: Yaml
The command line you used to run ctags:
```
ctags --options=NONE test.yaml
```

The content of input file:

```yml
en:
  test: 2
  other: 10
```

The tags output you are not satisfied with:

```
!_TAG_FILE_FORMAT	2	/extended format; --format=1 will not append ;" to lines/
!_TAG_FILE_SORTED	1	/0=unsorted, 1=sorted, 2=foldcase/
!_TAG_OUTPUT_EXCMD	mixed	/number, pattern, mixed, or combineV2/
!_TAG_OUTPUT_FILESEP	slash	/slash or backslash/
!_TAG_OUTPUT_MODE	u-ctags	/u-ctags or e-ctags/
!_TAG_PATTERN_LENGTH_LIMIT	96	/0 for no limit/
!_TAG_PROC_CWD	/home/akemrir/	//
!_TAG_PROGRAM_AUTHOR	Universal Ctags Team	//
!_TAG_PROGRAM_NAME	Universal Ctags	/Derived from Exuberant Ctags/
!_TAG_PROGRAM_URL	https://ctags.io/	/official site/
!_TAG_PROGRAM_VERSION	5.9.0	/p5.9.20220828.0/
```

The tags output you expect:

```
...
en.test	test.yaml	/^main(void)$/;"	kind:function	line:2	language:Yaml	signature:(void)	keys
en.other	test.yaml	/^main(void)$/;"	kind:function	line:3	language:Yaml	signature:(void)	keys
...

this could be tricky to omit one level?
...
test	test.yaml	/^main(void)$/;"	kind:function	line:2	language:Yaml	signature:(void)	keys
other	test.yaml	/^main(void)$/;"	kind:function	line:3	language:Yaml	signature:(void)	keys
...
```

The version of ctags:

```
$ ctags --version
Universal Ctags 5.9.0(p5.9.20220828.0), Copyright (C) 2015-2022 Universal Ctags Team
Universal Ctags is derived from Exuberant Ctags.
Exuberant Ctags 5.8, Copyright (C) 1996-2009 Darren Hiebert
  Compiled: Aug 29 2022, 21:05:02
  URL: https://ctags.io/
  Optional compiled features: +wildcards, +regex, +iconv, +option-directory, +xpath, +json, +interactive, +sandbox, +yaml, +packcc, +optscript, +pcre2
```

How do you get ctags binary:

Archlinux official repository
extra/ctags 1:5.9.20220828.0-1

---

I would have a map of keys paths from yaml file. This is for use with I18n ruby gem, for translations.
Then I could for example use it with vim to jump directly to translation and suggest path from vim addon.

Any hints to start off somewhere?
Maybe it would be similar to SASS?

### PR DESCRIPTION
**Pull request by masatake**
Close #3523

TODO:
- [x] revise dd52763.
- [ ] add a case giving unexpected input
- [x] add localeful extra (there may be cases that qualified extra doesn't fit)

### CODE REVIEW

**Comment by masatake** [comment]
I must add more comments to the change, [dd52763](https://github.com/universal-ctags/ctags/pull/3895/commits/dd5276377ddf37ba5af5198ddc65e2ea646b3559).

File: parsers/i18nrubygem.c [hunk 1]
```diff
@@ -0,0 +1,251 @@
+/*
+*
+*   Copyright (c) 2023, Masatake YAMATO
+*   Copyright (c) 20123, Red Hat, K.K.
```
Comment (akemrir) [inline parsers/i18nrubygem.c:4]: Is it typo?

Comment (masatake) [reply parsers/i18nrubygem.c:4]: Thank you.

**Comment by masatake** [comment]
[dae5d14](https://github.com/universal-ctags/ctags/pull/3895/commits/dae5d14886670a6081c9cf9ca88b06a38b9aa424) before merging this pull request, I must revise this commit.

**Comment by masatake** [comment]
@akemrir, can I ask you to reivew the new man page, [docs(man): add a page for I18nRubyGem](https://github.com/universal-ctags/ctags/pull/3895/commits/36a37b0434022f82d7b7b27b4635bdb0cac287f4) ?

**Comment by masatake** [comment]
As I was afraid, this is not useful for real-world use cases.

**Comment by masatake** [comment]
@akemrir, can I ask you to review the new man page again? [6f333c1](https://github.com/universal-ctags/ctags/pull/3895/commits/6f333c127419db1fbd9a403dab7765afb8088ae1)

**Comment by masatake** [comment]
@akemrir After I get an ack about the man page from you, I will merge this (with some modifications about [83eb3d3](https://github.com/universal-ctags/ctags/pull/3895/commits/83eb3d31386f264b38e1a8a29c6e22e06b1309e2)).

**Comment by masatake** [comment]
I could not get an ack but I will merge this for releasing 6.1.

**Pull request merged**